src/goto-symex/symex_goto.cpp

```diff
@@ -537,7 +539,13 @@ void goto_symext::symex_goto(statet &state)
         state.guard.add(guard_expr);
         new_state.guard.add(boolean_negate(guard_expr));
       }
+      auto new_state_guard_expr = new_state.guard.as_expr();
```

Comment:
:bulb: It would probably be more readable to make guard_exprt encapsulate the merge operation, but that's probably not possible due to the dependency on target. As a minimum, you could factor out the duplicate code into a helper function (similar to merge_state_guards).